### ISSUE TITLE
[P2] 機械学習の学習ループを視覚的に説明するデモページを実装する

### ISSUE THREAD

**Issue by tomiya7688** [tomiya7688/comfyUI_support_tools#92]
## 目的
数学的シミュレーションの一部として、機械学習の基本的な仕組みを**入力 → 特徴 → 重み付き計算 → 予測 → 誤差 → パラメータ更新 → 再予測**の循環として視覚的に理解できるデモページを実装する。

パーセプトロン単体の紹介ではなく、**「なぜ学習すると予測が変わるのか」**を、数式・パラメータ・グラフの変化を対応づけて見せることを目的とする。

このIssueでは高度なニューラルネットワークや実用MLフレームワークは扱わず、2変数程度の単純な二値分類を題材に、学習ループそのものを説明する。

## デモ対象
2次元入力 `x1, x2` と二値ラベル `0 / 1` の小規模データセットを使用する。

例:
```text
(x1, x2) -> label
(0.2, 0.3) -> 0
(0.8, 0.7) -> 1
...
```

内部モデルは理解しやすさを優先し、以下のような1ユニットの線形分類器でよい。

```text
z = w1*x1 + w2*x2 + b
prediction = activation(z)
```

学習では誤差に応じて `w1 / w2 / b` を少しずつ更新する。

## 実装内容
- [ ] 機械学習デモ専用ページを追加
- [ ] 2Dデータ点を散布図として表示
- [ ] クラス0 / 1を視覚的に区別する
- [ ] 現在の決定境界を線として表示
- [ ] 現在の `w1`, `w2`, `b` を表示
- [ ] 任意の1サンプルを選択し、順伝播の計算過程を表示
  - `x1 * w1`
  - `x2 * w2`
  - bias
  - 合計値 `z`
  - prediction
- [ ] 正解ラベルとpredictionの差を「誤差」として表示
- [ ] 学習stepを1回実行するボタンを追加
- [ ] 学習step後に `w1 / w2 / b` がどう変化したかを差分表示
- [ ] パラメータ更新に合わせて決定境界を再描画
- [ ] lossの履歴を簡単な折れ線グラフで表示
- [ ] 複数stepを自動実行できる簡易Run操作を追加
- [ ] Resetで初期状態へ戻せるようにする

## 視覚フロー
ページ上で最低限以下の流れを同時に確認できるようにする。

```text
Input data
   ↓
Weighted sum
   ↓
Prediction
   ↓
Compare with answer
   ↓
Error / Loss
   ↓
Update weights
   ↓
Decision boundary moves
```

可能であれば、現在処理している段階を矢印やハイライトで示す。

## UIイメージ
```text
[ Scatter Plot ]
  data points
  decision boundary

[ Current Sample ]
  x1 = 0.8
  x2 = 0.7

[ Calculation ]
  0.8 × w1(0.4) = 0.32
  0.7 × w2(-0.2) = -0.14
  bias = 0.10
  -----------------
  z = 0.28
  prediction = 0.57

answer = 1
loss = 0.43

[ Update ]
  w1: 0.40 -> 0.44
  w2: -0.20 -> -0.17
  b : 0.10 -> 0.15
```

## 説明文
各ブロックには短い説明を付ける。

例:
- 「入力値に重みを掛けて予測を作ります」
- 「予測と正解の差を誤差として測ります」
- 「誤差が小さくなる方向へ重みを少し変更します」
- 「これを繰り返すことで決定境界がデータに合っていきます」

説明は学術的な厳密さより、視覚デモとして理解しやすいことを優先する。

## テスト
- [ ] 初期パラメータで決定境界が表示される
- [ ] 1サンプルのweighted sumが正しく計算される
- [ ] prediction / lossが期待値と一致する
- [ ] 1 learning stepでパラメータが更新される
- [ ] パラメータ更新後に決定境界が変化する
- [ ] loss historyがstepごとに追加される
- [ ] Resetで初期状態へ戻る
- [ ] 同じ初期値とデータなら同じ学習結果を再現できる

## 完了条件
- 入力から予測が作られる過程を視覚的に追える
- predictionと正解の差からlossが生まれることを確認できる
- 1 stepの学習で重みが変わることを数値で確認できる
- 重みの変更に伴い決定境界が動くことを確認できる
- 複数stepでlossがどう変化するか確認できる
- 「機械学習は誤差を見ながらパラメータを調整する仕組み」という全体像がページ単体で伝わる

## Out of Scope
- 多層ニューラルネットワーク
- CNN / Transformer
- GPU学習
- PyTorch / TensorFlow等のframework統合
- 実データセット
- 高次元特徴量
- 本格的なoptimizer比較
- backpropagation computational graphの詳細表示
- 実運用向けモデル学習
